lopezMidterm/src/components/ProductDetails.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useFetch } from '../hooks/useFetch';
+import { useDispatch } from 'react-redux';
+import { addToCart } from '../actions/cartActions';
+import './ProductDetails.css';
+
+interface Product {
+  id: number;
+  title: string;
+  image: string;
+  description: string;
+}
+
+interface Error {
+  message: string;
+}
+
+const ProductDetails: React.FC = () => {
+  const { id } = useParams<{ id: string }>();
+  const { data: product, loading, error } = useFetch<Product>(`/products/${id}`);
+  const dispatch = useDispatch();
+
+  const handleAddToCart = () => {
+    if (product) {
+      dispatch(addToCart(product));
+      window.alert(`${product.title} added to cart`);
+    }
+  };
+
+  return (
+    <>
+      {loading ? (
```

Comment:
Return early like we have been doing in other exercises. Nested ternaries are tough to read and are prone to errors.

```
if (loading) return <Loading />
if (error) return <Error />

if (product) {
  // product JSX
} else {
  <p>No product found</p>
}
```